### ISSUE TITLE
What's the relationship to upstream WolfHUD?

### ISSUE THREAD

**Issue by vadcx** [steam-test1/VPlusHUD#281]
Hi, this is more of a management question, I want to know what I'd get into if I started contributing. I installed Vanilla Plus HUD myself and like it.

What I see is that (1) VPlusHUD is essentially a mod pack that bundles other mods, and (2) changed WolfHUD settings to look like it does, and (3) modified WolfHUD code for the same goal too?

https://github.com/Kamikaze94/WolfHUD/issues

I find it sad that this currently is a diverging hard fork. If I got into it,  I would like to help both, especially if you look at the WolfHUD open issues and maintainer's inactivity. This would require heavy refactoring of current code (I'm ready) to separate the config tweaks to WolfHUD as well as code changes, this would practically allow to rebase on top of WolfHUD, instead of running a completely custom codebase.

Although you've maintained this infinitely many times better, I guess there will still be bugs that are reported over there but not here. I wouldn't like to manually transfer fixes or track reports between the two projects. A single source would make life easier. Like here, a text without localization: [vplus](https://github.com/steam-test1/VPlusHUD/blob/b8d57176d4b1c81f91f16421b75d7e8689cb1168/devlua/TabStats.lua#L809), [wolfhud](https://github.com/Kamikaze94/WolfHUD/blob/1b2736f9290062fe73ef3f2f10b913255def4905/lua/TabStats.lua#L723).

The same applies to other mods. A clearer separation would allow for included 3rd party mod updates at a glance (like #256). To achieve this, I would try to persuade other mod authors (like WolfHUD) to add me as a maintainer or update their code (for example the crime.net performance fix - I don't fancy how it replaces the entire function, I believe a non-invasive override of the Steam APIs would fix the performance... while technically remaining `O(n^2)`)

Annotate code ripped from the game, potentially/eventually replace it to keep the repository clean.

---

My plan of action if you agree:

- refactor the code, to reunify with wolfhud
- ask to become a maintainer of wolfhud to take care of that side of the equation
- instead of replacing wolfhud functions where edits are needed for VPlusHUD, try to provide separated functions (to override), hooks from the WolfHUD side
- clearly separate into WolfHUD, VPlusHUD configs and code patches (additions), 3rd party mods
- contact mod authors to get the outdated versions updated (*to upstream* the accumulated 3rd party mod fixes)
- you would need to get more intimate with git and ideally have your local testing/development branch online to ease collaboration to simplify merging of pull requests

This would allow to bring fixes and features to both codebases and userbases with less effort in the future.

If you don't care or don't agree:

- do nothing :)
- contribute anyway
- start my own fork

---

TLDR: will you be accepting my invasive PRs or even put me in front of the wheel?
